modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -483,14 +507,18 @@ protected boolean usesDynamicConfiguration() {
    * returns an empty value.
    */
   protected Optional<ConfigurationInstance> getStaticConfiguration() {
-    if (!requiresConfig.get() || (isConfigurationSpecified() && configurationProvider.get().isDynamic())) {
+    if (!requiresConfig()) {
```

Comment:
isn't this changing the behavior?

Reply:
nope, not at all. On the contrary. this is not the first version of this optimization. This is the one that fixed all the tests that I broke :p